prime-router/src/test/kotlin/fhirengine/translation/HL7toFhirTranslatorTests.kt

```diff
@@ -81,17 +92,38 @@ DG1|1||F11.129^Opioid abuse with intoxication,unspecified^I10C|||W|||||||||1
 
     @Test
     fun `test birth date extension addition`() {
-        val message = HL7Reader(ActionLogger()).getMessages(supportedHL7ORM)
+        val message = HL7Reader(ActionLogger()).getMessages(supportedHL7ORMWithBirthDateTime)
+        assertThat(message.size).isEqualTo(1)
+        val bundle = HL7toFhirTranslator.getInstance().translate(message[0])
+        assertThat(bundle).isNotNull()
+        assertThat(bundle.type).isEqualTo(Bundle.BundleType.MESSAGE)
+        assertThat(bundle.id).isNotEmpty()
+        val patient = bundle.entry.first { it.resource.resourceType.name == "Patient" }.resource as Patient
+
+        val extensionValue = patient.birthDateElement.getExtensionByUrl(
+            "http://hl7.org/fhir/StructureDefinition/patient-birthTime"
+        ).value as DateTimeType
+
+        assertThat(
+            extensionValue.valueAsString
+        ).isEqualTo(Hl7RelatedGeneralUtils.dateTimeWithZoneId(birthDateTime, ""))
+    }
+
+    @Test
+    fun `test birth date extension is missing when birthdate is only date`() {
+        val message = HL7Reader(ActionLogger()).getMessages(supportedHL7ORMWithBirthDate)
         assertThat(message.size).isEqualTo(1)
         val bundle = HL7toFhirTranslator.getInstance().translate(message[0])
         assertThat(bundle).isNotNull()
         assertThat(bundle.type).isEqualTo(Bundle.BundleType.MESSAGE)
         assertThat(bundle.id).isNotEmpty()
+
         val patient = bundle.entry.first { it.resource.resourceType.name == "Patient" }.resource as Patient
+
         assertThat(
-            patient.getExtensionByUrl(
+            patient.birthDateElement.getExtensionByUrl(
                 "http://hl7.org/fhir/StructureDefinition/patient-birthTime"
-            ).valueAsPrimitive.toString()
-        ).isEqualTo(birthDate)
+            )
+        ).isNull()
```

Comment:
This line currently causes a compilation error.

Reply:
Oof, yep.  Import statements are hard.  Fixed.